### PR TITLE
Custom Markdown

### PR DESCRIPTION
**Pull request by msamper**
### Features and Changes

Introduces the possibility to display custom markdown on several key pages in GrowthBook to allow organizations to provide users with organization-specific guidance. 

Pages currently supported are:

1. Experiment List Page
2. Experiment Page
4. Feature List Page
5. Feature Page
6. Metric List Page
7. Metric Page

Custom Markdown users have the option to leverage handlebars variables available on each page and create templates tailored to the page/user. Every page has the following base variables available: `orgName` and `user`

The following pages have additional variables available for use:

- Experiment Page
  - `experiment`, `tags`, `experimentStatus`
- Feature Page
  - `featureKey`, `tags`, `featureType`
- Metric Page
  - `metricName`, `tags`, `metricType`, `metricDatasource`

### Testing

On an account with admin access, navigate to General Settings and access the Custom Markdown page at the bottom of the settings page. Try setting some markdown for the some of the pages and ensure that the new Custom Markdown component is rendered on the pages you set markdown for with the correct markdown.

### Screenshots

Screenshot 2024-07-18 at 12 06 16 PM
Screenshot 2024-07-18 at 12 06 23 PM
Screenshot 2024-07-18 at 12 11 53 PM

### CODE REVIEW

File: packages/front-end/pages/settings/custom-markdown.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import TempMessage from "@/components/TempMessage";
+import Button from "@/components/Button";
+import { useAuth } from "@/services/auth";
+
+const CustomMarkdown: React.FC = () => {
+  const { refreshOrganization, settings } = useUser();
+  const { apiCall } = useAuth();
+  const [saveMsg, setSaveMsg] = useState(false);
+
+  console.log({ settings });
```
Comment (jdorn) [inline packages/front-end/pages/settings/custom-markdown.tsx:17]: Remove this

File: packages/front-end/pages/settings/custom-markdown.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+              style={{ marginRight: "4rem" }}
+              color={"primary"}
+              //   disabled={!ctaEnabled}
+              onClick={async () => {
+                // if (!ctaEnabled) return;
+                await saveSettings();
+              }}
+            >
+              Save
+            </Button>
```
Comment (jdorn) [inline packages/front-end/pages/settings/custom-markdown.tsx:140]: Using the `<Modal>` component instead to handle form submits.

```tsx
<Modal inline submit={async () => {
  ...
}}>

</Modal>
```

File: packages/front-end/components/Markdown/CustomMarkdown.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+const CustomMarkdown: React.FC<Props> = ({ markdown, handlebarsVariables }) => {
+  if (!markdown) return null;
+
+  const template = Handlebars.compile(markdown);
+  const renderedMarkdown = template(handlebarsVariables);
+
+  return (
+    <div className="alert alert-info">
+      <Markdown>{renderedMarkdown}</Markdown>
```
Comment (jdorn) [inline packages/front-end/components/Markdown/CustomMarkdown.tsx:18]: Wrap in a div with max-height and `overflow-y: auto`

File: packages/front-end/components/Markdown/CustomMarkdown.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import Handlebars from "handlebars";
+import Markdown from "./Markdown";
+
+interface Props {
+  markdown?: string;
+  handlebarsVariables: Record<string, any>;
+}
+
+const CustomMarkdown: React.FC<Props> = ({ markdown, handlebarsVariables }) => {
```
Comment (jdorn) [inline packages/front-end/components/Markdown/CustomMarkdown.tsx:10]: Add a standard set of variables that are always available (user, orgName).  Only pass in additional ones as needed

**Review by Kevin-Chant — APPROVED** [review]
LGTM

**Pull request merged**